learn/appchains/tx-fees.md

```diff
@@ -0,0 +1,116 @@
+    - **WeightToFee** - a polynomial function that converts weight to fee
+- **Weight Fee** - fee defined by two separate parameters:
+    - **Benchmarked Weight** - weight that accounts for the complexity (execution time) of a specific call
+    - **Congestion Multiplier** - a function that converts weight to fee and can be adjusted to account for the congestion of the network (weight consumed in the previous block). For Tanssi Appchains, this is a constant
+- **Length Fee** - a function that defines the per-byte fee (_LenghtToFee_) correlated to the length in bytes of the function call. It can range from a constant to a cubic curve. For Tanssi Appchains, this is a constant
+- **Tip** - an optional value that increases the overall fee, increasing the priority of the transaction by incentivizing block producers to include it in the next block
+
+Therefore, in general terms, the transaction fee can be calculated with the following equation:
+
+<!-- https://github.com/moondance-labs/substrate/blob/master/frame/support/src/weights/extrinsic_weights.rs#L57 -->
+<!-- https://github.com/moondance-labs/tanssi/blob/master/container-chains/templates/frontier/runtime/src/lib.rs#L265-L277 -->
+
+```
+BaseFee = ExtrinsicBaseWeight * WeightToFee
+WeightFee = BenchmarkedWeight  * CongestionMultiplier
+LengthFee = ByteLenghtFunctionCall * LengthToFee
+
+InclusionFee = BaseFee + WeightFee + LenghtFee
+FinalFee = InclusionFee + Tip
+```
+
+All non-EVM function calls available to developers use these baseline calculations for transaction fees. EVM Tanssi Appchains have an extra layer to translate this fee scheme into an Ethereum-like scheme from an Ethereum JSON RPC and EVM perspective. 
+
+### EVM Transaction Fees
+
+<!-- https://github.com/polkadot-evm/frontier/blob/272fe8839f87161ed89350de166b379f1f4c6136/frame/base-fee/src/lib.rs#L126-L199 -->
+
+Tanssi offers [templates for full EVM-compatible Appchains](/builders/build/templates/evm/){target=\_blank}. Such Appchains provide an Ethereum-like environment for developers, where they can use Eth-specific libraries like [Ethers.js](/builders/interact/ethereum-api/libraries/ethersjs/){target=\_blank}, [Hardhat](/builders/interact/ethereum-api/dev-env/hardhat/){target=_blank}, and [Foundry](/builders/interact/ethereum-api/dev-env/foundry/){target=\_blank}.
+
+
+In addition, all Tansis EVM-compatible Appchains have an [EIP-1559 compatible](https://eips.ethereum.org/EIPS/eip-1559){target=\_blank} transaction pricing mechanism for EVM transactions. But they support both commonly used EVM transaction types:
+
+- **Type 0 (Legacy)** - the transaction fee is calculated through a single _gasPrice_ value that is included in the signed transaction blob. Because Tanssi EVM-compatible Appchains have a dynamic pricing mechanism, _gasPrice_ must be greater than the current block _baseFee_ for a transaction to be considered valid
```

Comment:
```suggestion
- **Type 0 (Legacy)** - the transaction fee is calculated through a single _gasPrice_ value that is included in the signed transaction blob. Because Tanssi EVM-compatible Appchains have a dynamic pricing mechanism, _gasPrice_ must be greater than the current block's _baseFee_ for a transaction to be considered valid
```